src/daft-micropartition/src/micropartition.rs

```diff
@@ -467,6 +468,166 @@ fn materialize_scan_task(
     Ok((table_values, cast_to_schema))
 }
 
+fn stream_scan_task(
```

Comment:
This is basically `materialize_scan_task` but streaming and only works with csv right now.